### PR TITLE
add getNextWorkRequired realisation for the VbkBlock

### PR DESCRIPTION
**Pull request by Mr-Leshiy**

### CODE REVIEW

File: include/veriblock/blockchain/block_index.hpp [hunk 1]
```diff
@@ -48,6 +48,23 @@ struct BlockIndex {
     return pbegin[(pend - pbegin) / 2];
   }
 
+  const BlockIndex* getAncestorBlocksBehind(height_t steps) const {
+    if (steps < 0 || steps > this->height) {
+      return nullptr;
+    }
+
+    if (steps == 0) {
```
Comment (blackyblack) [inline include/veriblock/blockchain/block_index.hpp]: Checks for steps are not necessary. Code will work the same way with an assumption that steps <= 0 return current block.

File: test/blockchain/vbk_blockchain_util_test.cpp [hunk 2]
```diff
@@ -0,0 +1,191 @@
...
+#include "veriblock/blockchain/block_index.hpp"
+#include "veriblock/blockchain/vbk_blockchain_util.hpp"
+#include "veriblock/blockchain/vbk_chain_params.hpp"
+
+using namespace VeriBlock;
+
+static std::vector<BlockIndex<VbkBlock>> getChain(int32_t deltaTime,
+                                                  int32_t deltaTime_change,
+                                                  const uint32_t& difficulty,
+                                                  const uint32_t& chainlength) {
```
Comment (blackyblack) [inline test/blockchain/vbk_blockchain_util_test.cpp]: chainlength and difficulty should be passed by value.

File: test/blockchain/vbk_blockchain_util_test.cpp [hunk 3]
```diff
@@ -0,0 +1,191 @@
...
+  BlockIndex<VbkBlock> blockIndex;
+  blockIndex.header.height = 1;
+  blockIndex.header.timestamp = 10000;
+  blockIndex.header.difficulty = difficulty;
+  blockIndex.height = blockIndex.header.height;
+  blockIndex.pprev = nullptr;
+
+  std::vector<BlockIndex<VbkBlock>> chain(chainlength);
+  chain[0] = blockIndex;
```
Comment (blackyblack) [inline test/blockchain/vbk_blockchain_util_test.cpp]: If chainlength is zero an exception will be thrown here. Put an assert to make sure chainlength is not zero.

File: test/blockchain/vbk_blockchain_util_test.cpp [hunk 4]
```diff
@@ -0,0 +1,191 @@
...
+  int32_t deltaTime = chainparams->getTargetBlockTime();
+
+  BlockIndex<VbkBlock> blockIndex;
+  blockIndex.header.height = 1;
+  blockIndex.header.timestamp = 10000;
+  blockIndex.header.difficulty = ArithUint256::fromHex("09184E72A000").toBits();
+  blockIndex.height = blockIndex.header.height;
+  blockIndex.pprev = nullptr;
+
+  std::vector<BlockIndex<VbkBlock>> chain(chainlength);
```
Comment (blackyblack) [inline test/blockchain/vbk_blockchain_util_test.cpp]: I think you could use getChain() here.

Comment (Mr-Leshiy) [reply test/blockchain/vbk_blockchain_util_test.cpp]: Here is used another algorithm of the chain building that is hard to unify to all purposes of the previous tests

File: include/veriblock/blockchain/block_index.hpp (hunk 1 above)
Comment (Mr-Leshiy) [reply include/veriblock/blockchain/block_index.hpp]: fixed

File: test/blockchain/vbk_blockchain_util_test.cpp (hunk 2 above)
Comment (Mr-Leshiy) [reply test/blockchain/vbk_blockchain_util_test.cpp]: fixed

File: test/blockchain/vbk_blockchain_util_test.cpp (hunk 3 above)
Comment (Mr-Leshiy) [reply test/blockchain/vbk_blockchain_util_test.cpp]: fixed

**Pull request merged**